core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -120,8 +119,10 @@ protected boolean resolveIndex() {
         final IndexMetaData metaData = primary.indexSettings().getIndexMetaData();
         Translog.Location location = null;
         for (int requestIndex = 0; requestIndex < request.items().length; requestIndex++) {
-            location = executeBulkItemRequest(metaData, primary, request, location, requestIndex,
+            if (request.items()[requestIndex].getPrimaryResponse() == null) {
```

Comment:
this is tricky - see [here](https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java#L209), which shows we may have a primary response, but still execute the item... I'm not sure it's wrong, I saying "watch out" ...